generator/src/main/java/com/scottlogic/deg/generator/VisualiseExecute.java

```diff
@@ -0,0 +1,121 @@
+package com.scottlogic.deg.generator;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.decisiontree.*;
+import com.scottlogic.deg.generator.decisiontree.visualisation.DecisionTreeVisualisationWriter;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpecFactory;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpecMerger;
+import com.scottlogic.deg.generator.fieldspecs.RowSpecMerger;
+import com.scottlogic.deg.generator.inputs.ProfileReader;
+import com.scottlogic.deg.generator.outputs.targets.OutputTarget;
+import com.scottlogic.deg.generator.reducer.ConstraintReducer;
+import com.scottlogic.deg.generator.validators.ErrorReporter;
+import com.scottlogic.deg.generator.validators.StaticContradictionDecisionTreeValidator;
+import com.scottlogic.deg.generator.validators.ValidationResult;
+import com.scottlogic.deg.generator.validators.VisualisationConfigValidator;
+import com.scottlogic.deg.generator.visualisation.VisualisationConfigSource;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * This class performs the visualisation of a profile and outputs the visualisation as a graphviz dot file..
+ */
+public class VisualiseExecute implements Runnable {
+
+    private final ErrorReporter errorReporter;
+    private final OutputTarget outputTarget;
+    private final ProfileReader profileReader;
+    private final VisualisationConfigSource configSource;
+    private final VisualisationConfigValidator validator;
+
+    @Inject
+    public VisualiseExecute(OutputTarget outputTarget,
+                            ProfileReader profileReader,
+                            VisualisationConfigSource configSource,
+                            VisualisationConfigValidator validator,
+                            ErrorReporter errorReporter) {
+        this.errorReporter = errorReporter;
+        this.configSource = configSource;
+        this.outputTarget = outputTarget;
+        this.profileReader = profileReader;
+        this.validator = validator;
+    }
+
+    @Override
+    public void run() {
+
+        final DecisionTreeFactory profileAnalyser = new ProfileDecisionTreeFactory();
```

Comment:
Shouldn't this be injected as a dependency?

Reply:
This issue was about adding '--overwrite' and refactoring ConfigSource. do we want to expand it into more refactoring?

Reply:
just checked the issue, and yes it was about everything in visualise.

Reply:
True, but you've refactored the `Visualise` type to use IoC, but not completely, its wierd for it to partly use IoC but not completely.